Assets/MixedRealityToolkit/Utilities/MixedRealityPlayspace.cs

```diff
@@ -0,0 +1,130 @@
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public static Transform Transform
+        {
+            get
+            {
+                if (mixedRealityPlayspace)
+                {
+                    return mixedRealityPlayspace;
+                }
+
+                if (CameraCache.Main.transform.parent == null)
+                {
+                    mixedRealityPlayspace = new GameObject(Name).transform;
+                    CameraCache.Main.transform.SetParent(mixedRealityPlayspace);
+                }
+                else
+                {
+                    if (CameraCache.Main.transform.parent.name != Name)
+                    {
+                        // Since the scene is set up with a different camera parent, its likely
+                        // that there's an expectation that that parent is going to be used for
+                        // something else. We print a warning to call out the fact that we're 
+                        // co-opting this object for use with teleporting and such, since that
+                        // might cause conflicts with the parent's intended purpose.
+                        Debug.LogWarning($"The Mixed Reality Toolkit expected the camera\'s parent to be named {Name}. The existing parent will be renamed and used instead.");
+                        // If we rename it, we make it clearer that why it's being teleported around at runtime.
+                        CameraCache.Main.transform.parent.name = Name;
+                    }
+
+                    mixedRealityPlayspace = CameraCache.Main.transform.parent;
+                }
+
+                // It's very important that the Playspace align with the tracked space,
+                // otherwise reality-locked things like playspace boundaries won't be aligned properly.
+                // For now, we'll just assume that when the playspace is first initialized, the
+                // tracked space origin overlaps with the world space origin. If a platform ever does
+                // something else (i.e, placing the lower left hand corner of the tracked space at world 
+                // space 0,0,0), we should compensate for that here.
+                return mixedRealityPlayspace;
+            }
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public static Vector3 Position => Transform.position;
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public static Quaternion Rotation => Transform.rotation;
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="transform"></param>
+        public static void AddChild(Transform transform)
+        {
+            transform.SetParent(Transform);
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public static void DetachChildren()
+        {
+            Transform.DetachChildren();
+        }
```

Comment:
The API surface exposed here feels a bit interesting - there's a public one to add a specific child (i.e. as a consumer I can invoke that to add a specific child) but there's no way for me to remove that child without totally messing up all other callers (i.e. I can't remove just my child, I have to remove everything).

I'm guessing that maybe this could be cleared up with a comment, or perhaps is coming in the next PR, but this API assymetry should be cleared up.

Reply:
This is based, a bit on Unity's transform API. there is a RemoveChildAt(index), but no RemoveChild(transform).

@WiWei, can you file an issue to add better symetry?

Reply:
I guess what I don't understand is, is my question even a problem (i.e. will Detach children ever be called by multiple people, or just a single person managing some greater lifecycle?) I'm trying to understand the calling mechanics of this function.

---
In reply to: [279107558](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/4099#discussion_r279107558) [](ancestors = 279107558)

Reply:
it >might< be called by multiple people. i will remove for now until more thinking can be applied